index/scorch/persister.go

```diff
@@ -789,20 +800,24 @@ var NumSnapshotsToKeep = 1
 
 // Removes enough snapshots from the rootBolt so that the
 // s.eligibleForRemoval stays under the NumSnapshotsToKeep policy.
-func (s *Scorch) removeOldBoltSnapshots() (numRemoved int, err error) {
+func (s *Scorch) removeOldBoltSnapshots(forceCompact bool) (numRemoved int, err error) {
 	persistedEpochs, err := s.RootBoltSnapshotEpochs()
 	if err != nil {
 		return 0, err
 	}
 
-	if len(persistedEpochs) <= s.numSnapshotsToKeep {
+	numSnapshotsToKeep := s.numSnapshotsToKeep
+	if forceCompact {
+		numSnapshotsToKeep = 1
```

Comment:
i think i'd prefer instead more "powerful" API's rather than passing in forceCompact which feels like a special case?  like the caller might just pass in numSnapshotsToKeep as the parameter rather than forceCompact?

e.g., more like... if the incoming numSnapshotsToKeep param is 0, then perhaps default to s.numSnapshotsToKeep?